bitswap.go

```diff
@@ -325,56 +332,91 @@ func (bs *Bitswap) ReceiveMessage(ctx context.Context, p peer.ID, incoming bsmsg
 		return
 	}
 
-	wg := sync.WaitGroup{}
-	for _, block := range iblocks {
-
-		wg.Add(1)
-		go func(b blocks.Block) { // TODO: this probably doesnt need to be a goroutine...
-			defer wg.Done()
-
-			bs.updateReceiveCounters(b)
-			bs.sm.UpdateReceiveCounters(p, b)
```

Comment:
Note: `bs.sm.UpdateReceiveCounters(p, b)` has been removed in favour of passing blocks *and* dups to `bs.sm.UpdateReceiveBlocksFrom(p, b)`, which now also takes care of accounting

Reply:
See also https://github.com/ipfs/go-bitswap/issues/156